pkg/repositories/handle.go

```diff
@@ -24,6 +24,9 @@ func NewDBHandle(dbConfigValues config.DbConfig, catalogScope promutils.Scope) (
 		User:         dbConfigValues.User,
 		Password:     dbConfigValues.Password,
 		ExtraOptions: dbConfigValues.ExtraOptions,
+		BaseConfig: config.BaseConfig{
+			DisableForeignKeyConstraintWhenMigrating: true,
```

Comment:
why is this true?

Reply:
So I believe that we do not populate foreign keys at all using GORM v1. Therefore I turn off populating foreign key during migration in GORM v2 to match the behavior of v1. I think we can address that in a separate PR.

Here are the table constraints populated by v1 gorm and you can see no foreign constraint is created.
```
datacatalog=> select table_name, constraint_name, constraint_type from information_schema.table_constraints where table_name in ('tags', 'artifacts', 'datasets');
 table_name |    constraint_name    | constraint_type
------------+-----------------------+-----------------
 datasets   | datasets_pkey         | PRIMARY KEY
 datasets   | datasets_uuid_key     | UNIQUE
 tags       | tags_pkey             | PRIMARY KEY
 artifacts  | artifacts_pkey        | PRIMARY KEY
 artifacts  | 2200_16421_4_not_null | CHECK
 artifacts  | 2200_16421_5_not_null | CHECK
 artifacts  | 2200_16421_6_not_null | CHECK
 artifacts  | 2200_16421_7_not_null | CHECK
 artifacts  | 2200_16421_8_not_null | CHECK
 datasets   | 2200_16412_4_not_null | CHECK
 datasets   | 2200_16412_5_not_null | CHECK
 datasets   | 2200_16412_6_not_null | CHECK
 datasets   | 2200_16412_7_not_null | CHECK
 tags       | 2200_16439_4_not_null | CHECK
 tags       | 2200_16439_5_not_null | CHECK
 tags       | 2200_16439_6_not_null | CHECK
 tags       | 2200_16439_7_not_null | CHECK
 tags       | 2200_16439_8_not_null | CHECK
 tags       | 2200_16439_9_not_null | CHECK
(19 rows)
```

Reply:
ah interesting, mind filing an issue in https://github.com/flyteorg/flyte to explore enabling it in a separate PR?